### PR TITLE
UGENE-6579 Intersect annotations does not work if an annotation group consists of digits only

### PR DESCRIPTION
**Pull request by rasputinkirill**
Test(s) will be added after successful code review.

### CODE REVIEW

File: src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp [hunk 1]
```diff
@@ -179,6 +181,9 @@ QList<Task *> BedtoolsIntersectAnnotationsByEntityTask::onSubTaskFinished(Task *
         Document *resultDoc = loadResultTask->getDocument();
         CHECK_EXT(resultDoc != NULL, setError(tr("Result document is NULL")), res);
         result = resultDoc->findGObjectByType(GObjectTypes::ANNOTATION_TABLE);
+        AnnotationTableObject *ato = qobject_cast<AnnotationTableObject *>(result.first());
```
Comment (yalgaer) [inline src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp:185]: Check that result list is not empty before calling first() ?

File: src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp [hunk 2]
```diff
@@ -179,6 +181,9 @@ QList<Task *> BedtoolsIntersectAnnotationsByEntityTask::onSubTaskFinished(Task *
...
+        CHECK_EXT(ato != NULL, setError(tr("Failed to cast to AnnotationTableObject")), res);
```
Comment (yalgaer) [inline src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp]: Here you failed to find annotation object, not to cast. 

findGObjectByType(GObjectTypes::ANNOTATION_TABLE) guarantees that you have a valid AnnotationTableObjects in the list.

File: src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp [hunk 3]
```diff
@@ -179,6 +181,9 @@ QList<Task *> BedtoolsIntersectAnnotationsByEntityTask::onSubTaskFinished(Task *
...
+        renameAnnotationsFromBed(ato->getRootGroup());
```
Comment (yalgaer) [inline src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp:186]: Process all annotation table objects from the list (result)?

File: src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp [hunk 4]
```diff
@@ -199,11 +204,32 @@ Document *BedtoolsIntersectAnnotationsByEntityTask::createAnnotationsDocument(co
...
     return doc;
 }
 
+void BedtoolsIntersectAnnotationsByEntityTask::renameAnnotationsForBed(AnnotationGroup *group) {
+    bool isNumber = false;
+    group->getName().toInt(&isNumber);
+    if (isNumber) {
+        group->setName(group->getName() + RENAME_STRING);
+    }
+    foreach(AnnotationGroup * g, group->getSubgroups()) {
```
Comment (yalgaer) [inline src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp]: use for, not foreach

File: src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.h [hunk 5]
```diff
@@ -139,6 +142,8 @@ class BedtoolsIntersectAnnotationsByEntityTask : public ExternalToolSupportTask
     SaveMultipleDocuments *saveAnnotationsTask;
     BedtoolsIntersectTask *intersectTask;
     LoadDocumentTask *loadResultTask;
+
+    QMap<QString, int> renameMap;
```
Comment (yalgaer) [inline src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.h]: Delete unused field

File: src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp [hunk 6]
```diff
@@ -199,11 +204,32 @@ Document *BedtoolsIntersectAnnotationsByEntityTask::createAnnotationsDocument(co
...
+        renameAnnotationsForBed(g);
+    }
+}
+
+void BedtoolsIntersectAnnotationsByEntityTask::renameAnnotationsFromBed(AnnotationGroup *group) {
+    if (group->getName().contains(RENAME_STRING)) {
```
Comment (yalgaer) [inline src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp]: Use endsWith()

**Review by yalgaer — CHANGES_REQUESTED** [review]

File: src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp (hunk 3 above)
Comment (rasputinkirill) [reply src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp:186]: According to knowledge how to bedtools works there is only one annotation table in the resulting file.

File: src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp (hunk 4 above)
Comment (rasputinkirill) [reply src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp]: Why? What's wrong with foreach?

**Review by yalgaer — APPROVED** [review]

File: src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp (hunk 4 above)
Comment (yalgaer) [reply src/plugins/external_tool_support/src/bedtools/BedtoolsIntersectTask.cpp]: foreach (Qt only) was added when there was no for (C++)

Compare
 foreach(AnnotationGroup * g, group->getSubgroups()) {
vs
 for(AnnotationGroup * g: group->getSubgroups()) {

Tools that are not aware about QT may not understand foreach macro (refactoring, highlighting, formatting, etc...)

**Pull request merged**